GetAssetTrigger/index.ts

```diff
@@ -37,34 +37,51 @@ import { toNetError } from "../lib/net/to_net_err"
  *         description: The Asset could not be found.
  */
 const GetAssetTrigger: AzureFunction = async function (context: Context, req: HttpRequest): Promise<void> {
-  const { id } = context.bindingData
   const logger = makeLogger(context.log)
   const collibraClient = await makeCollibraClient(req.headers.authorization)(logger)
 
   const res = await pipe(
-    getAssetByID(collibraClient)(id),
-    TE.bindTo("collibraAsset"),
-    TE.bind("approvedStatus", () => getStatusByName(collibraClient)("Approved")),
-    TE.bind("approvedAsset", ({ approvedStatus, collibraAsset }) =>
-      pipe(
-        TE.fromEither(determineAssetStatus(approvedStatus.name as string)(collibraAsset)),
-        TE.mapLeft(() => toNetError(403)("This asset is not approved"))
-      )
-    ),
-    TE.bind("collibraAttributes", () => getAssetAttributes(collibraClient)(id)),
-    TE.bind("collibraCommunity", ({ approvedAsset }) =>
-      getCommunityByDomainID(collibraClient)(approvedAsset.domain.id)
-    ),
-    TE.bind("tags", ({ approvedAsset }) => getAssetTags(collibraClient)(approvedAsset.id)),
-    TE.map(({ approvedAsset, collibraAttributes, collibraCommunity, tags }) =>
-      assetAdapter(collibraAttributes)(collibraCommunity)(tags)(approvedAsset)
+    context.bindingData.id,
+    isValidID,
+    E.mapLeft(toNetError(400)),
+    TE.fromEither,
+    TE.chain(getAsset(collibraClient)),
+    TE.mapLeft(toNetError(500)),
+    TE.map(
+      (asset) =>
```

Comment:
The map function might be a separate function to make this code a little bit shorter and more readable